model/riscv_insts_zce_ext.sail

```diff
@@ -0,0 +1,1021 @@
+                         X(rd) = EXTZ(X(rd)[15..0]);
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
+}
+
+mapping clause assembly = C_LHU(uimm, rsc, rdc)
+  <-> "c.lhu" ^ spc() ^ creg_name(rdc) ^ sep() ^ creg_name(rsc) ^ sep() ^ hex_bits_4(uimm @ 0b000)
+/* ****************************************************************** */
+union clause ast = C_LH : (bits(1), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ _ C0]
+mapping clause encdec_compressed = C_LH(ui5, rs1, rd)
+  <-> 0b100 @ 0b001 @ rs1 : cregidx @ 0b1 @ ui5 : bits(1) @ rd : cregidx @ 0b00
+
+function clause execute (C_LH(uimm, rsc, rdc)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+  
+  let rd = creg2reg_idx(rdc);
+  let rs = creg2reg_idx(rsc);
+
+  let rsext = X(rs) + immext;
+  X(rs) = rsext;
+  
+   match execute(LOAD(imm, rs, rd, false, WORD, false, false)) {
+     RETIRE_SUCCESS => {
+                         X(rd) = EXTS(X(rd)[15..0]);
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
+}
+
+mapping clause assembly = C_LH(uimm, rsc, rdc)
+  <-> "c.lh" ^ spc() ^ creg_name(rdc) ^ sep() ^ creg_name(rsc) ^ sep() ^ hex_bits_4(uimm @ 0b000)
+/* ****************************************************************** */
+
+union clause ast = C_SB : (bits(2), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ C0]
+mapping clause encdec_compressed = C_SB(ui65, rs1, rs2)
+  <-> 0b100 @ 0b010 @ rs1 : cregidx @ ui65 : bits(2) @ rs2 : cregidx @ 0b00
+
+function clause execute (C_SB(uimm, rs1c, rs2c)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+
+  let rs1c = creg2reg_idx(rs1c);
+  let rs2c = creg2reg_idx(rs2c);
+
+  let rsext = X(rs1c) + immext;
+  
+  let rs2c_lbyte = X(rs2c)[7..0];
+
+  X(rs1c) = rsext;
+  //X(rs2c) = EXTZ(rs2c_lbyte);
+  
+  X(rs2c) = EXTZ(X(rs2c)[7..0]);
+  //mem(rsext[7..0]) = X(rs2c);
+  
+  match execute(STORE(imm, rs2c, rs1c, WORD, false, false)){
+     RETIRE_SUCCESS => RETIRE_SUCCESS,
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
+}
+
+mapping clause assembly = C_SB(uimm, rs1, rs2)
+  <-> "c.sb" ^ spc() ^ creg_name(rs2) ^ sep() ^ creg_name(rs1) ^ sep() ^ hex_bits_5(uimm @ 0b000)
+/* ****************************************************************** */
+union clause ast = C_SH : (bits(1), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ C0]
+mapping clause encdec_compressed = C_SH(ui5, rs1, rs2)
+  <-> 0b100 @ 0b011 @ rs1 : cregidx @ 0b0 @ ui5 : bits(1) @ rs2 : cregidx @ 0b00
+
+function clause execute (C_SH(uimm, rs1c, rs2c)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+
+  let rs2c = creg2reg_idx(rs2c);
+  let rs1c = creg2reg_idx(rs1c);
+
+  let rsext = X(rs1c) + immext;
+  //let rs2c_lbyte = X(rs2c)[15..0];
+
+  X(rs2c) = EXTZ(X(rs2c)[15..0]);
+
+  // MemValue(rsext[15..0]) = X(rs2c);
+  match execute(STORE(imm, rs2c, rs1c, WORD, false, false)){
```

Comment:
WORD instead of HALF ? same as the comments on C_SB ?

Reply:
Same comments as above